### PR TITLE
Add security section to enterprise guide in docs.

### PR DESCRIPTION
**Pull request by mandiwise**
This PR adds a new section to the Enterprise Guide in the docs about securing a federated graph.

### CODE REVIEW

**Review by StephenBarlow — APPROVED** [review]
Eek I forgot about this! It looks good!

File: docs/source/enterprise-guide/graph-security.mdx [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+const gateway = new ApolloGateway({
+  // ...
+  buildService({ name, url }) {
+    const fetcher = (input, init) => {
+      if (init) {
+        init.timeout = 3000;
+      } else {
+        init = { timeout: 3000 };
+      }
+      return fetch(input, init);
```
Comment (abernix) [inline docs/source/enterprise-guide/graph-security.mdx:134]: Is it left to an exercise to the user to provide this from some Fetch API?  (I don't see it noted that it's an exercise to the user? 🤔 )

File: docs/source/enterprise-guide/graph-security.mdx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+## Authentication and authorization
+
+There is no one "right" way to handle access control with GraphQL, and consolidated data graphs are no exception to this. How authentication and authorization are handled for requests to any APIs you have in use currently will likely influence how you approach these concerns with federation as well.
+
+However, one key question to address with federation is the matter of where you will validate a token sent from a client. For example, you may use middleware to intercept and validate a token in a gateway Apollo Server, add it to the request object, and then get that validated token and add it to the Apollo Server `context`:
+
+```javascript
+const server = ApolloServer({
+  gateway,
+  subscriptions: false,
```
Comment (abernix) [inline docs/source/enterprise-guide/graph-security.mdx:214]: This line is no longer necessary in Apollo Server 3.0!

```suggestion
```

File: docs/source/enterprise-guide/graph-security.mdx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+### Know who’s using your graph (and how)
+
+To enhance the utility of traces collected in your observability tooling, it's a best practice to require clients to identify themselves and that they consistently name operations. The web and mobile versions of Apollo Client make it easy to set custom headers for the name and version of a specific client to [segment traces in Apollo Studio](https://www.apollographql.com/docs/studio/client-awareness/and metrics by client). Other API clients can set the `apollographql-client-name` and `apollographql-client-version` request headers manually to provide client awareness as well. (As a bonus, client awareness also makes it easier to identify what clients may be impacted by a proposed breaking change to your API when running schema checks.)
```
Comment (abernix) [inline docs/source/enterprise-guide/graph-security.mdx:248]: There's a syntax error that I've fixed, but I'm not sure if the sentence is what you expected it to be.

```suggestion
To enhance the utility of traces collected in your observability tooling, it's a best practice to require clients to identify themselves and that they consistently name operations. The web and mobile versions of Apollo Client make it easy to set custom headers for the name and version of a specific client to [segment traces and metrics in Apollo Studio by client](https://www.apollographql.com/docs/studio/client-awareness/). Other API clients can set the `apollographql-client-name` and `apollographql-client-version` request headers manually to provide client awareness as well. (As a bonus, client awareness also makes it easier to identify what clients may be impacted by a proposed breaking change to your API when running schema checks.)
```

File: docs/source/enterprise-guide/graph-security.mdx [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+Once there are measures in place to limit API discoverability in public-facing environments, the next step in protecting a GraphQL API is to guard it against both intentionally and unintentionally malicious queries. Again, many GraphQL-related vulnerabilities have to do with how an unprotected API may be exploited in DoS attacks, but there are other considerations as well. 
+
+In the sections that follow, we will explore measures that can help mitigate the impact of malicious queries for any GraphQL API, such as limiting query depth and amount, paginating list fields where appropriate, validating and sanitizing data, setting timeouts, and guarding against batched query abuse. And for GraphQL APIs with third-party clients, we will also explore using query cost analysis to support rate-limiting.
+
+#### Limit query depth
+
+GraphQL allows clients to traverse through a data graph and express complex relationships between the nodes in an operation's selection set. But as far as backing data sources are concerned, this can quickly turn into too much of a good thing when there are no guardrails to restrict how deeply queries can be nested. For example:
+
+```graphql
+query {
```
Comment (abernix) [inline docs/source/enterprise-guide/graph-security.mdx:57]: Bit pedantic of me, but trying to align with our [operation best practices](https://www.apollographql.com/docs/react/data/operation-best-practices/) to always provide operation names. 😉 

```suggestion
query DeepBlogQuery {
```

Feel free to reject this!

**Review by abernix — APPROVED** [review]
Left a few comments and a couple suggestions but this looks amazing!  Thanks so much!

File: docs/source/enterprise-guide/graph-security.mdx (hunk 2 above)
Comment (mandiwise) [reply docs/source/enterprise-guide/graph-security.mdx]: Good catch, thanks.

File: docs/source/enterprise-guide/graph-security.mdx (hunk 1 above)
Comment (mandiwise) [reply docs/source/enterprise-guide/graph-security.mdx:137]: Thanks for flagging. I've added import lines to improve clarity.

File: docs/source/enterprise-guide/graph-security.mdx (hunk 4 above)
Comment (mandiwise) [reply docs/source/enterprise-guide/graph-security.mdx]: Added, thanks.

File: docs/source/enterprise-guide/graph-security.mdx (hunk 3 above)
Comment (mandiwise) [reply docs/source/enterprise-guide/graph-security.mdx]: Good catch, fixed now.

**Pull request merged**